nexus/sdk-python.md

```diff
@@ -0,0 +1,330 @@
+            # ...
+            return EchoOutput(input.message)
+
+        return SyncOperation(start)
+```
+
+This isn’t attractive or Pythonic. Firstly, inline function `def`s, where necessary, should be short in Python; our users shouldn’t be forced to implement their own complex application logic in a function `def` inlined in a method body. Secondly, users are going to want to give `start` a `self` first parameter, but having that together with the service instance `self` that it’s closing over is much too confusing. Even if they get the `self`-lessness right, it would leave the `start` method defined inline not matching the signature that it “normally” has when defined on an operation class.
+
+<img width="1314" alt="Image" src="https://github.com/user-attachments/assets/6c2aee0a-f1be-47ed-bfb5-8e738304a418" />
+
+
+# Temporal Nexus
+
+Currently, the SDK functionality that we’re implementing related to Temporal is (a) the ability for workflow callers to start, cancel, and receive the results from Nexus operations, and (b) on the handler side, a Nexus worker that receives (non-durable) Nexus tasks and routes them to the requested verb handler for the requested operation in the requested service.
+
+The operations may be implemented as described above (although with support for `start` and `cancel` only; status is obtained from caller-side in-workflow futures, and async results are delivered via the Nexus protocol callback, which is a purely server-side concern).
+
+Alternatively, the author of a Temporal Nexus service may choose to define an always-async operation backed by a Temporal execution primitive. Workflow is the only such backing primitive we will launch with, but we plan to offer operations backed by updates, signals, and probably our top-level activity/task and other future features, when they exist.
+
+It’s the “always-async operation backed by a Temporal execution primitive” that we’re discussing in this section. We’re going to provide a convenience shorthand, analogous to the one above for defining sync operations, that implements a workflow-backed operation in an opinionated way. For users who want an unopinionated interaction with workflows and other Temporal primitives, a Temporal client will be available via a `ContextVar` in the vanilla `start` methods and they can do whatever they want with it, as long as they return their result synchronously (this has a `10s` limit). In contrast, our utility will do the right thing w.r.t. e.g. request-ID-based deduplication, and will be the only way to implement an arbitrary-duration async operation in a Temporal Nexus service.
+
+The proposal here follows the  [2b. Convenience utilities for shorthand operation definition](#2b-convenience-utilities-for-shorthand-operation-definition) `@nexusrpc.handler.sync_operation` decorator above:
+
+```python
+@nexusrpc.handler.service(interface=interface.MyNexusService)
+class MyNexusService:
+    @temporalio.nexus.handler.workflow_operation
```

Comment:
Safe to assume this is just shorthand for something like:

```python
def hello(self) -> WorkflowOperationHandler[HelloInput, HelloOutput]:
    return temporalio.nexus.handler.WorkflowOperationHandler(HelloWorkflow.run)
```

And I could do that myself if I didn't want the decorator (e.g. maybe doing something a bit more dynamic)?